changed build definitions such that we can publish to sonatype.

marcelluethi opened the pull request:
As bintray sunsetting its services, we will now publish to sonatype (maven central).
This PR makes the necessary changes to the build definitions. Note that we also publish the api doc, as in conrast to bintray, we cannot publish a package without api doc.

Commit:
changed build definitions such that we can publish to sonatype.

Andreas-Forster reviewed (approved):
To me, it looks reasonable. Thank you a lot for your effort in figuring out how to set this up properly.

Commit:
change definitions for manual deployment

It turns out that automatic deployment to Sonatype does not
work at the current stage, due to the fixup step. This commit
therefore changes to build definitions such, that publishing is
done to a tmp directory.

marcelluethi commented:
> 
> 
> To me, it looks reasonable. Thank you a lot for your effort in figuring out how to set this up properly.

It looks reasonable but it still turned out to be wrong, as the fixup command needs to be applied locally. 
The latest commit that I added changes the build definition such that it is published to a tmp directory, from which we can manually sign it and upload it to sonatype.

Andreas-Forster commented:
Great that you found the problem and a fix for it in this short time. What about the manual process of publishing it after the building and local publishing processes happened, should we document that somewhere or would it even be possible to add a fixAndPublish.sh script to the library?
